projects/lights_control/inc/lights_gpio.h

```diff
@@ -1,53 +1,55 @@
 #pragma once
+// This module abstracts the operation on gpio pins. It simplifies control of gpio pins relating to
+// various outputs by accepting LIGHTS_EVENT_GPIO_* events with the data field set to the
+// specific peripheral.
 
 #include "event_queue.h"
 #include "gpio.h"
 #include "status.h"
 
-// lights_gpio requires lights_gpio_config to be initialized first.
+#include "lights_events.h"
 
-// This module abstracts the operation on gpio pins. It simplifies control of various peripherals
-// by relating them to events. It needs to be initialized first to set up all the gpio pins.
-
-// Used for making peripheral maps
-#define LIGHTS_GPIO_PERIPHERAL_BIT(peripheral_index) ((1) << (peripheral_index))
+// Used for making output bit-sets.
+#define LIGHTS_GPIO_OUTPUT_BIT(output_index) ((1) << (output_index))
 
+// State definitions.
 typedef enum {
   LIGHTS_GPIO_STATE_OFF = 0,
   LIGHTS_GPIO_STATE_ON,
   NUM_LIGHTS_GPIO_STATES
 } LightsGPIOState;
 
+// Polarity definitions.
 typedef enum {
   LIGHTS_GPIO_POLARITY_ACTIVE_HIGH,
   LIGHTS_GPIO_POLARITY_ACTIVE_LOW,
   NUM_LIGHTS_GPIO_POLARITIES
 } LightsGPIOPolarity;
 
-// Bitset where every bit maps to a peripheral.
-typedef uint16_t LightsGPIOPeripheralMapping;
+// Bit-set where every bit maps to an output.
+typedef uint16_t LightsGPIOOutputBitset;
 
-typedef struct LigthsGPIOPeripheral {
+typedef struct LightsGPIOOutput {
```

Comment:
Please rename your instances of `GPIO` to `Gpio`. We're planning on moving all of our acronym naming to only capitalize the first letter (see ELEC-334) for context/example. Essentially, something like `CANUART` would be replaced with `CanUart`.